include/agent.h

```diff
@@ -0,0 +1,73 @@
+#ifndef _ABM_AGENT_H_
+#define _ABM_AGENT_H_
+
+#include <memory>
+#include <vector>
+
+#include "agent.h"
+#include "config.h"
+#include "graph.h"
+
+namespace abm {
+
+//! Agent class that has id, origin, destination, departure time, current node
+//! and compute path from current node to destination.
+class Agent {
+ public:
+  //! Construct an agent class from either an id, or id, origin and destination
+  //! \param[in] id Agent id
+  //! \param[in] origin Agent origin
+  //! \param[in] destination Agent destination
+  explicit Agent(const graph::vertex_t& id) : id_{id} {};
```

Comment:
Please only declare functions in the header, do not define them. Please define functions in the implementation .cc file

Reply:
But I see `{ return nvertices_; }` in `graph.h`